File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let ans = 0;
    const visited = new Array(n).fill(false);
    function dfs(node){
        visited[node] = true;
        for(let i = 0; i < n; i++){
            if(computers[node][i] === 1 && !visited[i]) dfs(i);
        }
    }
    
    for(let i = 0; i < n; i++){
        if(!visited[i]){
            dfs(i)
            ans++;
        }
    }
    console.log('q')
    return ans;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    visited = [False] * n
    answer = 0
    
    def dfs(cur):
        visited[cur] = True
        for nxt in range(n):
            if computers[cur][nxt] == 1 and not visited[nxt]:
                dfs(nxt)
                
    for i in range(n):
        if not visited[i]:
            dfs(i)
            answer += 1
    return answer




